integration/containerd/confidential/agent_api.bats

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bats
+# Copyright (c) 2022 Red Hat
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/lib.sh"
+load "${BATS_TEST_DIRNAME}/asserts.sh"
+load "${BATS_TEST_DIRNAME}/../../../lib/common.bash"
+
+setup() {
+	start_date=$(date +"%Y-%m-%d %H:%M:%S")
+	sandbox_name="kata-cc-busybox-sandbox"
+	pod_config="${FIXTURES_DIR}/pod-config.yaml"
+	pod_id=""
+
+	echo "Delete any existing ${sandbox_name} pod"
+	crictl_delete_cc_pod_if_exists "$sandbox_name"
+
+	echo "Prepare containerd for Confidential Container"
+	SAVED_CONTAINERD_CONF_FILE="/etc/containerd/config.toml.$$"
+	configure_cc_containerd "$SAVED_CONTAINERD_CONF_FILE"
+
+	echo "Reconfigure Kata Containers"
+	clear_kernel_params
+	switch_image_service_offload on
```

Comment:
As this code matches the `agent_image.bats` I wonder if there is merit in moving it to a shared library to avoid the duplication?

Reply:
Yeah, It does make sense to remove that duplication. I wish it was Python (or any OO language) where we could use heritage. Mind if I send it in a follow up PR?